api/v1alpha/api.proto

```diff
@@ -37,6 +37,114 @@ enum ImageType {
+        INSECURE_OPTION_ON_DISK = 8;
+        INSECURE_OPTION_HTTP    = 16;
+        INSECURE_OPTION_PUBKEY  = 32;
+}
+
+// DiscoverOption defines the discovery options to be used for fetching.
+enum DiscoverOption {
+        DISCOVER_OPTION_NONE       = 0;
+        DISCOVER_OPTION_NO_STORE   = 1;
+        DISCOVER_OPTION_ONLY_STORE = 2;
+}
+
+// FetchImageRequest defines the request for FetchImage.
+message FetchImageRequest {
+        // Type of the image, e.g. aci, docker, oci, etc.
+        Image type = 1;
+
+        // Resolvable name of the image, used for discovery, can be string, hash, URL, file path.
+        string name = 2;
+
+        // Insecure options for fetch, it's a bitflag.
+        InsecureOption insecure_option = 3;
+
+        // Whether to fetch from the store, or skip the store.
+        DiscoverOption discover_option = 4;
+
+        // Whether to fetch the depedencies.
+        bool with_deps = 5;
+
+        // Serialized credential config file, used for fetching.
+        // The format is documented at:
+        // https://github.com/coreos/rkt/blob/master/Documentation/configuration.md#rktkind-auth
+        bytes credentials_confg = 6;
+
+        // TODO(yifan): Add discover scheme. Or add them in DiscoverOption?
+}
+
+// FetchImageResponse defines the response for FetchImage.
+message FetchImageResponse {
+        // ID of the fetched image, which can be used to uniquely identify the image.
+        string id = 1;
+}
+
+// PreparePodRequest defines the request for PreparePod.
+message PreparePodRequest {
+        // The pod manifest to use for preparing.
+        bytes pod_manifest = 1;
+}
+
+// PreparePodResponse defines the response for PreparePod.
+message PreparePodResponse {
+        // ID of the prepared rkt pod, which can be used to invoke `rkt run-prepared`.
+        string id = 1;
+}
+
+// RemovePodRequest defines the request for RemovePod.
+message RemovePodRequest {
+        // ID of the pod to remove, when specified, at most one image will be removed.
+        string id = 1;
+}
+
+// RemovePodResponse defines the response for RemovePod.
+message RemovePodResponse {}
+
+// RemoveImageRequest defines the request for RemoveImage.
+message RemoveImageRequest {
+        // ID of the image to remove, when specified, at most one image will be removed.
+        string id = 1;
+
+        // Name of the image to remove, if multiple images have the same name,
+        // all of them will be removed.
+        string name = 2;
+}
+
+// RemoveImageResponse defines the response for RemoveImage.
+message RemoveImageResponse {}
+
+// EnterPodRequest defines the request for EnterPod.
+message EnterPodRequest {
+        // ID of the pod to enter.
+        string id = 1;
+
+        // Name of the app inside the pod to enter.
+        // This must be non-empty if there are multiple apps in the pod.
+        string app_name = 2;
+
+        // The command to run for enter, optional.
+        repeated string command = 3;
```

Comment:
@euank I think there's some difference:
- `command` is used to invoke a process in the namespace when the service receives the rpc for the first time.
- `stdin` is a bunch of bytes(strings) that send to the process launched above.